### PR TITLE
Expand `dask.utils.is_arraylike` docstring.

### LINKED ISSUE #7327: Rename `is_arraylike` to `is_duck_array`
Based on some recent discussion, in particular this comment ( https://github.com/dask/dask/pull/7321#issuecomment-791243195 ), it would be good to rename `is_arraylike` to `is_duck_array` for clarity and possibly update the docstring of this function to reflect its intended usage.

cc @shoyer @pentschev @jrbourbeau

### PR DESCRIPTION
**Pull request by douglasdavis**
- [x] Closes #7327 
- [ ] Tests added / passed
- [x] Passes `black dask` / `flake8 dask`

Specifically to point out difference between what the function is checking and how NumPy defines `array_like`

### CODE REVIEW

**Comment by jakirkham** [comment]
cc @shoyer @pentschev @jrbourbeau @jsignell @mrocklin (in case any of you have thoughts/suggestions on this change)

File: dask/utils.py [hunk 1]
```diff
@@ -1111,6 +1111,17 @@ def partial_by_order(*args, **kwargs):
 def is_arraylike(x):
     """Is this object a numpy array or something similar?
 
+    This function tests specifically for an object that already has
+    array attributes (e.g. np.ndarray, dask.array.Array), **NOT** for
```
Comment (jrbourbeau) [inline dask/utils.py:1115]: We might add `cupy.ndarray` and `sparse.COO` arrays as two additional examples to broaden readers context beyond NumPy

**Review by shoyer — APPROVED** [review]
Looks good to me, thanks!

File: dask/utils.py [hunk 2]
```diff
@@ -1111,6 +1111,17 @@ def partial_by_order(*args, **kwargs):
...
+    something that can be coerced into an array object (e.g. Python
+    lists and tuples). It is meant for dask developers and developers
+    of downstream libraries.
+
+    Note that this function is inconsistent with NumPy's definition of
```
Comment (jrbourbeau) [inline dask/utils.py:1120]: This has a slightly more positive connotation

```suggestion
    Note that this function does not correspond with NumPy's definition of
```

**Review by jrbourbeau — COMMENTED** [review]
Thanks @douglasdavis! I left a couple of small, non-blocking comments

File: dask/utils.py (hunk 2 above)
Comment (douglasdavis) [reply dask/utils.py:1120]: Agreed :+1:

**Comment by douglasdavis** [comment]
thanks @jrbourbeau changes incoprorated :+1:

File: dask/utils.py [hunk 3]
```diff
@@ -1111,6 +1111,17 @@ def partial_by_order(*args, **kwargs):
 def is_arraylike(x):
     """Is this object a numpy array or something similar?
 
+    This function tests specifically for an object that already has
+    array attributes (e.g. np.ndarray, dask.array.Array, cupy.ndarray,
+    sparse.COO), **NOT** for something that can be coerced into an
+    array object (e.g. Python lists and tuples). It is meant for dask
+    developers and developers of downstream libraries.
+
+    Note that this function is does not correspond with NumPy's
```
Comment (pentschev) [inline dask/utils.py:1120]:
```suggestion
    Note that this function does not correspond with NumPy's
```
```suggestion
    Note that this function is does not correspond with NumPy's
```

**Review by pentschev — COMMENTED** [review]
Looks good, suggested a minor typo fix, thanks @douglasdavis !

File: dask/utils.py (hunk 3 above)
Comment (jrbourbeau) [reply dask/utils.py:1120]: Whoops, that's my fault -- thanks for the update @pentschev

**Review by jrbourbeau — APPROVED** [review]
Thanks @douglasdavis! This is in

**Pull request merged**